lib/cli/src/commands/create_exe.rs

```diff
@@ -223,14 +224,31 @@ impl CreateExe {
             return Err(anyhow::anyhow!("input path cannot be a directory"));
         }
 
-        let (_, compiler_type) = self.compiler.get_store_for_target(target.clone())?;
+        let (_, ct) = self.compiler.get_store_for_target(target.clone())?;
+
+        let compiler_type = if self.compiler.singlepass {
+            CompilerType::Singlepass
+        } else if self.compiler.cranelift {
+            CompilerType::Cranelift
+        } else if self.compiler.llvm {
+            CompilerType::LLVM
+        } else {
+            ct // default compiler type
+        };
+
+        let engine = EngineBuilder::new(get_config(&compiler_type)?).engine();
+
+        let tunables = BaseTunables::for_target(engine.target());
+        let store = Store::new_with_tunables(&engine, tunables);
```

Comment:
I'm a bit confused... I think the store returned by `get_store_for_target` already does this?

Am I missing something? Have you tried doing this?
```suggestion
        let (store, ct) = self.compiler.get_store_for_target(target.clone())?;
```

And if you did, and it's different... can you explore on why your own-constructed `Store` is different from the one returned by `get_store_for_target` ?

Reply:
@syrusakbary the store returned by `get_store_for_target` emits the wrong `UnwindInfo` type on Windows (windows-native, but we need DWARF unwind info, even on Windows). I need to do `Cranelift::default()`, `Singlepass::default()`, etc. instead. The `get_store_for_target` doesn't work for `create-exe`.

Reply:
> the store returned by `get_store_for_target` emits the wrong UnwindInfo type on Windows (windows-native, but we need DWARF unwind info, even on Windows)

I don't think this is true. I believe you were using `Store::default()` somewhere else and that caused a mismatch when using the store generated by `get_store_for_target`.
In any case, Can you explain what's the difference between your code and `get_store_for_target`? Basically, and following your assumption, I'd like to know why `get_store_for_target` is generating wrong unwind info and your code is not

Reply:
`Store::default()` used a mix between the host-native and the target flags / features.